pybatchai.py

```diff
@@ -0,0 +1,84 @@
+import logging
+
+from azure.common.credentials import ServicePrincipalCredentials
+import click
+import coloredlogs
+
+import cli.resource_group
+import cli.validation
+
+@click.group()
+@click.option('--subscription-id', required=True, 
+              callback=cli.validation.validate_uuid)
+@click.option('--resource-group', required=True, help='resource group name', 
+              callback=cli.validation.validate_rg_name)
```

Comment:
more nitpicking!!
I would move the help= to a newline
```
@click.option('--resource-group', required=True,
              help='resource group name', 
              callback=cli.validation.validate_rg_name)
```
but looking at this I would probably also want to alphabetize it
```
@click.option('--resource-group', required=True, 
              callback=cli.validation.validate_rg_name,
              help='resource group name')
``` 
which has the additional coincidental advantage of organizing by length.

at this point I think I'm just giving into my OCD; this code is totally fine.

Reply:
In honor of js :p I put callback at the end, since typically callbacks are the last param of a multi param function. (that's why I didn't alphabetize it)

Also, I don't think there's a need for a newline as its < 80 chars and still is pretty readable.